### ISSUE TITLE
Generic "pt" vs  "pt-PT" and "pt-BR"

### ISSUE THREAD

**Issue by lidel** [ipfs/i18n#7]
Now and then, someone requests adding `pt-BR` "Portuguese (Brazil)" to our translation projects at Transifex.

Until now I believed country-specific variants of Portuguese do not differ as much as Chinese ones, so it was more efficient to deduplicate efforts and focus on a single, generic locale: 

- https://www.transifex.com/ipfs/dashboard/all_projects/pt/

I don't speak Portuguese myself, so hereby I summon some native speakers active in IPFS community (including managers of js.ipfs.io website)  to weight on this :-)

@satazor @hugomrdias @PedroMiguelSS @hacdias @fsdiogo  @pgte @daviddias 
@agentofuser @dntxos

### Question:

- Are differences between `pt-PT` and `pt-BR` in written language big enough to justify duplicated efforts?
  - Would be useful to provide examples of existing translations that would look different in `pt-BR`
  - Perhaps its just a few labels that we could work around by creating a generic glossary?

**Comment by agentofuser** [comment]
Copying over my comments from private conversation with @lidel on transifex:

```
agentofuser wrote:
> Hi lidel,
> 
> Thanks for getting in touch. I saw the pt translation
> and it's good that there is something already. There
> are though some pretty significant differences between
> pt_BR and pt_PT, especially in colloquial language
> (different pronouns and conjugations) and technical
> terms (Brazilians won't know what a "ficheiro" is; we
> call files "arquivos"). I'm sure there can be some
> reuse between the translations, but adoption in Brazil
> will be much easier with a proper pt_BR translation
> that feels familiar. I'll be more than glad to help
> with that.
```

**Comment by hacdias** [comment]
@lidel, yes, what @agentofuser mentioned is true (the "files" translation is one of the huge differences) but there are more. Perhaps we should support both yes.

**Comment by hugomrdias** [comment]
yeah we should support both

**Comment by PedroMiguelSS** [comment]
Yes, that makes sense indeed. We should support pt-BR as well.

**Comment by lidel** [comment]
Thanks, we have a consensus then :)

DONE: 

- [x] added `pt-BR` to existing projects, so @agentofuser  and others can start translating
- [x] updated `lang_map` in `.tx/config` in all projects so `tx pull -a` (locale sync) will create proper filenames with `pt-BR` instead of `pt_BR`

TODO:

I kept `pt` as-is for now, as renaming it to `pt_PT` requires additional work:

- [ ] Rename `pt`→`pt_PT` at Transifex
  - Crude way to do it is to add `pt_PT`, download translations from `pt` and upload them to `pt_PT`, then delete `pt`
  -  [ ] I emailed Transifex if they could just rename `pt` to `pt_PT` in all out projects (will update this issue when I get the response)
- [ ] Rename hardcoded `pt`→`pt_PT` in existing projects
    - js.ipfs.io, Web UI, share app etc hardcode lang codes in additional config files. This means I need to rewire each app to use `pt_PT` instead of `pt` in a separate PR

**Comment by agentofuser** [comment]
Well I'm glad I dug that hole for myself now 😅 IPFS Companion translation is done, going to jump to IPFS Desktop next unless there's a higher priority I should know of.
